File: frontend/app/src/main/java/com/example/easypark/easyparkfinal/session/SessionManager.java
package com.example.easypark.easyparkfinal.session;

import android.content.SharedPreferences;

public class SessionManager {

    private static SessionManager instance;
    private ISession session;
    private SharedPreferences sharedPreferences;

    public static SessionManager getInstance(){
        if(instance == null){
            instance = new SessionManager();
        }
        return instance;

    }

    public SharedPreferences getSharedPreferences() {
        return sharedPreferences;
    }

    public void setSharedPreferences(SharedPreferences sharedPreferences) {
        this.sharedPreferences = sharedPreferences;
    }

    public ISession getSession() {
        return session;
    }
    public void setSession(ISession session){
        this.session = session;
    }

    public ISession createSession(SharedPreferences shared){
        this.sharedPreferences =shared;
        session = new ISession() {
            @Override
            public boolean isLoggedIn() {
                return sharedPreferences.contains("token");
            }

            @Override
            public void saveToken(String token) {
                SharedPreferences.Editor ed = sharedPreferences.edit();
                ed.putString("token", token);
                ed.commit();
            }

            @Override
            public String getToken() {
                return sharedPreferences.getString("token","DEFAULT");
            }

            @Override
            public void saveEmail(String email) {
                SharedPreferences.Editor ed = sharedPreferences.edit();
                ed.putString("email", email);
                ed.commit();
            }

            @Override
            public String getEmail() {
                return sharedPreferences.getString("email","DEFAULT");
            }

            @Override
            public Long getId() {
                return sharedPreferences.getLong("id",0);
            }

            @Override
            public void saveId(Long id) {
                SharedPreferences.Editor ed = sharedPreferences.edit();
                ed.putLong("id", id);
                ed.commit();
            }

            @Override
            public void savePassword(String password) {
                SharedPreferences.Editor ed = sharedPreferences.edit();
                ed.putString("senha", password);
                ed.commit();
            }

            @Override
            public String getPassword() {
                return sharedPreferences.getString("senha","DEFAULT");
            }

            @Override
            public void invalidate() {
                SharedPreferences.Editor ed = sharedPreferences.edit();
                ed.remove("senha");
                ed.remove("email");
                ed.remove("token");
                ed.remove("id");
                ed.commit();
            }
        };
        return session;
    }
}

File: MobileTruck/app/src/main/java/easypark/com/mobiletruck/session/SessionManager.kt
package easypark.com.mobiletruck.session


import android.content.SharedPreferences

class SessionManager {

    lateinit var session: Session
    lateinit var sharedPreferences: SharedPreferences



    fun createSession(sharedPreferences: SharedPreferences){
        this.sharedPreferences = sharedPreferences
        this.session = object : Session{
            override fun getId(): Long {
                return sharedPreferences.getLong("id",0)

            }

            override fun saveId(id: Long) {
                sharedPreferences.edit()
                        .putLong("id",id).apply()
            }

            override fun saveToken(token: String) {
                sharedPreferences.edit()
                        .putString("token",token).apply()
            }

            override fun getToken(): String? {
                return sharedPreferences.getString("token","DEFAULT")
            }

            override fun saveEmail(email: String) {
                sharedPreferences.edit()
                        .putString("email",email).apply()
            }

            override fun getEmail(): String? {
                return sharedPreferences.getString("email","DEFAULT")
            }

            override fun savePassword(password: String) {
                sharedPreferences.edit()
                        .putString("senha",password).apply()
            }

            override fun getPassword(): String? {
                return sharedPreferences.getString("senha","DEFAULT")
            }

            override fun invalidate() {
                sharedPreferences.edit().remove("email")
                        .remove("senha")
                        .remove("token").apply()
            }

            override fun isLoggedIn(): Boolean {
                return sharedPreferences.contains("token")
            }
        }
    }
    companion object{
        val instance = SessionManager()
    }

}